controllers/rollingupgrade_controller.go

```diff
@@ -463,14 +463,15 @@ func (r *RollingUpgradeReconciler) populateAsg(ruObj *upgrademgrv1alpha1.Rolling
 	return nil
 }
 
-func (r *RollingUpgradeReconciler) populateLaunchTemplates() error {
+func (r *RollingUpgradeReconciler) populateLaunchTemplates(ruObj *upgrademgrv1alpha1.RollingUpgrade) error {
 	launchTemplates := []*ec2.LaunchTemplate{}
 	err := r.EC2Client.DescribeLaunchTemplatesPages(&ec2.DescribeLaunchTemplatesInput{}, func(page *ec2.DescribeLaunchTemplatesOutput, lastPage bool) bool {
 		launchTemplates = append(launchTemplates, page.LaunchTemplates...)
 		return page.NextToken != nil
 	})
 	if err != nil {
-		return err
+		r.error(ruObj, err, "Failed to populate launch template list")
+		return errors.Wrap(err, ruObj.Name+": Failed to populate launch template list")
```

Comment:
can use `fmt.Errorf`
```suggestion
		return fmt.Errorf("failed to populate launch template list for %q: %w", ruObj.Name, err)
```

Reply:
@adammw , let's commit this suggestion so that we can go ahead and merge this change

Reply:
any reason to use that over the error logger? it appears to give a better result as it'll automatically add the stacktrace and other niceities

Reply:
sorry, thought this was a comment on the logger line not error wrapping. updated.